docs/current_docs/integrations/snippets/php/go/main.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"context"
+	"fmt"
+)
+
+type MyModule struct{}
+
+// return container image with application source code and dependencies
+func (m *MyModule) Build(source *Directory) *Container {
+	return dag.Container().
+		From("php:8.2-apache-buster").
+		WithExec([]string{"apt-get", "update"}).
+		WithExec([]string{"apt-get", "install", "--yes", "git-core", "zip", "curl"}).
+		WithExec([]string{"docker-php-ext-install", "pdo", "pdo_mysql", "mysqli"}).
+		WithExec([]string{"sh", "-c", "sed -ri -e 's!/var/www/html!/var/www/public!g' /etc/apache2/sites-available/*.conf"}).
+		WithExec([]string{"sh", "-c", "sed -ri -e 's!/var/www/!/var/www/public!g' /etc/apache2/apache2.conf /etc/apache2/conf-available/*.conf"}).
+		WithExec([]string{"a2enmod", "rewrite"}).
+		WithDirectory("/var/www", source.WithoutDirectory("dagger"), ContainerWithDirectoryOpts{
+			Owner: "www-data",
+		}).
+		WithWorkdir("/var/www").
+		WithExec([]string{"chmod", "-R", "775", "/var/www"}).
+		WithExec([]string{"sh", "-c", "curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer"}).
+		WithMountedCache("/root/.composer", dag.CacheVolume("composer-cache")).
```

Comment:
@marcosnils even though I have cache volumes enabled, I don't see much difference when re-running `dagger call build --source=.` Any idea why?

Reply:
Hey Vikram! I do see quite a difference. 

When I have the cache volumes enabled, I don't see the `composer install` installing the modules multiple times. 

without cache volumes I get this output every time when I add a `composer.json` with any dependency

```
dagger -v call build --source .

  ┃ Writing lock file                                                                                      
  ┃ Installing dependencies from lock file (including require-dev)                                         
  ┃ Package operations: 2 installs, 0 updates, 0 removals                                                  
  ┃   - Downloading psr/log (3.0.0)                                                                        
  ┃   - Downloading monolog/monolog (3.6.0)                                                                
  ┃  0/2 [>---------------------------]   0%                                                               
  ┃  1/2 [==============>-------------]  50%                                                               
  ┃  2/2 [============================] 100%                                                               
  ┃   - Installing psr/log (3.0.0): Extracting archive                                                     
  ┃   - Installing monolog/monolog (3.6.0): Extracting archive                                             
  ┃  0/2 [>---------------------------]   0%                                                               
  ┃  2/2 [============================] 100%                                                               
  ┃ 11 package suggestions were added by new dependencies, use `composer suggest` to see details.          
  ┃ Generating autoload files                                                                              
  ┃ 1 package you are using is looking for funding.                                                        
  ┃ Use the `composer fund` command to find out more! 
``` 

if I add the cache volumes, that `install` phase only happens once.

Reply:
@vikram-dagger have another input that will make this run considerably faster. If you move the `composer install` a bit above, before the `WithDirectory` call, that will speed things up given that the step won't be executed each time the `source` changes.

Reply:
> @vikram-dagger have another input that will make this run considerably faster. If you move the `composer install` a bit above, before the `WithDirectory` call, that will speed things up given that the step won't be executed each time the `source` changes.

`composer install` needs the `composer.json` file from the source, which becomes available only after calling `WithDirectory`. I tried it to confirm and it failed. Not sure if I maybe misunderstood your suggestion?